designs/Remote-Authentication.md

```diff
@@ -0,0 +1,45 @@
+# Remote Authentication
+
+* Status: ** Draft **
+* Author(s): [Rob Relyea](https://github.com/rrelyea)
+
+## Issue
+
+TODO
+
+## Problem Background
+
+Several operations require authentication to a source in NuGet:
+
+- restore
+- search
+- icon fetching for search
+- update
+
+There are 3 authentication techniques in NuGet.
+
+ 1. VS Auth - Inside VS, NuGet can demand implementers of Auth. Azure Artifacts ships one with VS.
+ 1. NuGet CLI v1 Auth - NuGet.exe historically allowed an exe in the same directory that nuget.exe would communicate with.
+ 1. NuGet CLI v2 Auth - In NuGet 4.8 (VS 15.8), we enabled a new xplat auth plugin technique. This allows xplat auth in all of our codepaths (nuget.exe, msbuild.exe, dotnet.exe and VS).
+
+ When solving this problem when there is some nuget code running on the client and server, we need to decide which of those to support.
+
+## Possible approaches
+
+1. Remote v2 Auth call to client - When credentials are needed, the server calls to the client to authenticate via prompts, etc... Relies on NuGet CLI v2 Auth provider being installed on VS client side.
+2. support VS auth providers too? - not sure if this parallel code base is important to support.

```

Comment:
I think it's up to us. 
We'll need a server side coalescing NuGet provider and given that VS supports V2 providers too, we can do it with only 1 NuGet provider if we wanted to.